network/src/main/scala/com/linkedin/norbert/network/partitioned/loadbalancer/DefaultLoadBalancerHelper.scala

```diff
@@ -101,16 +94,12 @@ trait DefaultLoadBalancerHelper extends LoadBalancerHelpers with Logging {
           loopCount = loopCount + 1
         } while (loopCount <= es)
 
-        compensateCounter(idx, loopCount, counter);
-        return Some(endpoints(idx % es).node)
+        compensateCounter(idx, loopCount, counter)
+        Some(endpoints(idx % es).node)
     }
   }
 
-  def compensateCounter(idx: Int, count:Int, counter:AtomicInteger) {
-    if (idx + 1 + count <= 0) {
-      // Integer overflow
-      counter.set(idx + 1 - java.lang.Integer.MAX_VALUE + count)
-    }
-    counter.set(idx + 1 + count)
+  def isEndpointViable(capability: Option[Long], persistentCapability: Option[Long], endpoint: Endpoint): Boolean = {
```

Comment:
def's can be overriden with dynamic results, this way might actually be better (and a val is going to be a getter function in java anyway)